patches/0005-Integrate-OpenSSL-module.patch

```diff
@@ -0,0 +1,897 @@
+-	"crypto/internal/boring/bbig"
++	boring "crypto/internal/backend"
++	"crypto/internal/backend/bbig"
++	"crypto/internal/backend/bcache"
+ 	"math/big"
+ 	"unsafe"
+ )
+@@ -26,8 +27,8 @@ import (
+ // still matches before using the cached key. The theory is that the real
+ // operations are significantly more expensive than the comparison.
+ 
+-var pubCache boring.Cache
+-var privCache boring.Cache
++var pubCache bcache.Cache
++var privCache bcache.Cache
+ 
+ func init() {
+ 	pubCache.Register()
+diff --git a/src/crypto/ecdsa/ecdsa.go b/src/crypto/ecdsa/ecdsa.go
+index 7ce7542872c..e12e8410eee 100644
+--- a/src/crypto/ecdsa/ecdsa.go
++++ b/src/crypto/ecdsa/ecdsa.go
+@@ -24,14 +24,14 @@ import (
+ 	"crypto/aes"
+ 	"crypto/cipher"
+ 	"crypto/elliptic"
+-	"crypto/internal/boring/bbig"
++	"crypto/internal/backend/bbig"
+ 	"crypto/internal/randutil"
+ 	"crypto/sha512"
+ 	"errors"
+ 	"io"
+ 	"math/big"
+ 
+-	"crypto/internal/boring"
++	boring "crypto/internal/backend"
+ 
+ 	"golang.org/x/crypto/cryptobyte"
+ 	"golang.org/x/crypto/cryptobyte/asn1"
+diff --git a/src/crypto/ecdsa/notboring.go b/src/crypto/ecdsa/notboring.go
+index 039bd82ed21..3cc16ecab56 100644
+--- a/src/crypto/ecdsa/notboring.go
++++ b/src/crypto/ecdsa/notboring.go
+@@ -2,11 +2,11 @@
+ // Use of this source code is governed by a BSD-style
+ // license that can be found in the LICENSE file.
+ 
+-//go:build !boringcrypto
++//go:build !boringcrypto && !goexperiment.opensslcrypto
+ 
+ package ecdsa
+ 
+-import "crypto/internal/boring"
++import boring "crypto/internal/backend"
+ 
+ func boringPublicKey(*PublicKey) (*boring.PublicKeyECDSA, error) {
+ 	panic("boringcrypto: not available")
+diff --git a/src/crypto/ed25519/ed25519_test.go b/src/crypto/ed25519/ed25519_test.go
+index 7c5181788f6..bf29780070c 100644
+--- a/src/crypto/ed25519/ed25519_test.go
++++ b/src/crypto/ed25519/ed25519_test.go
+@@ -9,7 +9,7 @@ import (
+ 	"bytes"
+ 	"compress/gzip"
+ 	"crypto"
+-	"crypto/internal/boring"
++	boring "crypto/internal/backend"
+ 	"crypto/rand"
+ 	"encoding/hex"
+ 	"os"
+diff --git a/src/crypto/hmac/hmac.go b/src/crypto/hmac/hmac.go
+index 34805765d55..79fd58d0da4 100644
+--- a/src/crypto/hmac/hmac.go
++++ b/src/crypto/hmac/hmac.go
+@@ -26,7 +26,7 @@ import (
+ 	"hash"
+ )
+ 
+-import "crypto/internal/boring"
++import boring "crypto/internal/backend"
+ 
+ // FIPS 198-1:
+ // https://csrc.nist.gov/publications/fips/fips198-1/FIPS-198-1_final.pdf
+diff --git a/src/crypto/hmac/hmac_test.go b/src/crypto/hmac/hmac_test.go
+index 55415abf020..904925377bb 100644
+--- a/src/crypto/hmac/hmac_test.go
++++ b/src/crypto/hmac/hmac_test.go
+@@ -6,7 +6,7 @@ package hmac
+ 
+ import (
+ 	"bytes"
+-	"crypto/internal/boring"
++	boring "crypto/internal/backend"
+ 	"crypto/md5"
+ 	"crypto/sha1"
+ 	"crypto/sha256"
+diff --git a/src/crypto/internal/boring/cache.go b/src/crypto/internal/boring/cache.go
+deleted file mode 100644
+index 476e47706c9..00000000000
+--- a/src/crypto/internal/boring/cache.go
++++ /dev/null
```

Comment:
I like the idea of putting this change into its own file. It will also be clearer which refinements we are doing to existing boring package for the sake of reusability.

Reply:
Done! Git does count it as a rename, so we saved almost 500 lines of diffs and future headaches if the original files we changed 😄